drivers/SerialBase.cpp

```diff
@@ -161,6 +153,26 @@ void SerialBase:: unlock()
     // Stub
 }
 
+void SerialBase::init(void)
+{
+    lock();
```

Comment:
Just to save a little ROM, I think I'd leave the locks out here. They're not needed in the constructor and destructor, and `UARTSerial` has them dummied anyway.

If you're going to be calling `init` or `deinit` you need more sequencing than just the basic `lock` anyway - imagine if someone just decided to do some more serial stuff after `deinit` did its `unlock`!

(Maybe stick a `_` in front of the method name to give a clue that they're unlocked methods. Some things use that convention, not sure if the serial stuff does).

Let me know if that doesn't make sense to you - I'm not quite 100% confident.